src/systems/CameraSystem.ts

```diff
@@ -1,27 +1,44 @@
 import type { ECSSystem } from '@/ecs/ECSSystem';
-import { PlayerBrand, type Player } from '@/entity/components/Player';
-import { PositionBrand, type Position } from '@/entity/components/Position';
+import { CameraBrand, type Camera } from '@/entity/components/Camera';
 import {
-  RenderableBrand,
-  type Renderable
-} from '@/entity/components/Renderable';
+  PositionBrand,
+  type Position,
+  hasPosition
+} from '@/entity/components/Position';
 import { SCALE } from '@/renderer/createGameRenderer';
 import type { RenderableId } from '@/renderer/renderableCache';
+import { addVector } from '@/utils/vectors';
 import type { Application, DisplayObject } from 'pixi.js';
+import { isDefined } from '@/utils/assertions';
 
 export const CameraSystem: (
   resolveSprite: (sprite: RenderableId) => DisplayObject,
   app: Application
-) => ECSSystem<[Player, Position, Renderable]> = (resolveSprite, app) => ({
-  target: [PlayerBrand, PositionBrand, RenderableBrand],
+) => ECSSystem<[Camera, Position]> = (resolveSprite, app) => ({
+  target: [CameraBrand, PositionBrand],
   run: (ecs, props, entities) => {
-    const player = entities[0];
-    if (!app || !player) return;
+    const [e] = entities;
+    if (!e) return;
 
```

Comment:
see my comment regarding camera instance number, there should always be only one active camera, right? I mean we could have many cameras, but then we should select by "active"